docusaurus/docs/iOS/06-advanced/06-chat-integration.mdx

```diff
@@ -5,32 +5,52 @@ description: How to integrate chat & video
 
 ## Introduction
 
-:::warning
-Still to add:
-
-- reference to the starter projects
-- explanation of the StreamWrapper class
-
-:::
-
 It's common for calling apps to have chat, as well as the opposite - chat apps to have a calling functionality. Stream's Chat and Video SDKs are perfectly compatible between each other, and can easily be integrated into an app.
 
 :::tip
-You can find example integrations, for both our UIKit and SwiftUI SDKs, in our sample apps [repository](https://github.com/GetStream/stream-video-ios-examples).
+You can find an example chat integration with our video product, in our samples [repo](https://github.com/GetStream/stream-video-ios-examples/tree/main/VideoWithChat/VideoWithChat).
 :::
 
-In the samples, we provide a `StreamWrapper` class, that handles the creation of the different clients and UI objects for both chat and video.
-
 ## Adding chat into video
 
 In this guide you will take a video-based application and add chat functionality with the Stream Chat SDK on top of it. Here is an example of what the end result will look like:
 
-### TODO: embed video here
+![Watch the gif](../assets/chat-integration.gif)
 
 :::info
-The starting point for this guide is a functioning video calling application. If you don't have one and want to follow along, feel free to do our [step-by-step tutorial](../../basics/tutorial) first.
+The starting point for this guide is a functioning video calling application. If you don't have one and want to follow along, feel free to do our [step-by-step tutorial](https://getstream.io/video/docs/ios/tutorials/video-calling/) first.
 :::
 
+### Connecting the Video and Chat Client
+
+You should have a setup for both the video and chat client. They both require the same API key, as well as the same user token.
+
+The sample repo contains a simple [`StreamWrapper`](https://github.com/GetStream/stream-video-ios-examples/blob/main/VideoWithChat/VideoWithChat/StreamWrapper.swift), that can help you initialize and connect both clients.
+
+Here's an example on how to initialize the `StreamWrapper`:
+
+```swift
+let streamWrapper = StreamWrapper(
+    apiKey: "YOUR_API_KEY",
+    userCredentials: user,
+    tokenProvider: { result in
+        Task {
+            do {
+                let token = try await TokenService.shared.fetchToken(for: user.id)
+                userRepository.save(token: token.rawValue)
+                result(.success(token.rawValue))
+            } catch {
+                result(.failure(error))
+            }
+        }
```

Comment:
Isn't this specific to our implementation? What about something more generic?
```suggestion
        let token = \\ Fetch the token from your backend
        result(.success(token))
```

Reply:
makes sense, I updated it 👍